feat: upgrade graphql-upload dependency 13.0.0 -> 16.0.1

mmilardic opened the pull request:
Upgraded graphql-upload to 16.0.1 and made other necessary changes

vuki656 commented:
Few questions:

- Why all the eslint disables?
- Also, why move `graphql-upload` from dev to prod dep?
- Why remove `@types/graphql-upload` dep?

Sidenote, you could update all other deps here (non major ones)

mmilardic commented:
> 

- Because eslint won't allow .mjs imports. Wouldn't bother doing more than this if the issue is introduces only by the file extension.
- Removed it from both dev and peer dependencies and moved to prod. Not sure why it was in the dev dependencies in the first place.

Other dependencies/peerdependencies minor version were upgraded in last commit.

I wouldn't put more effort into this if it works as is now. There is no plan to maintain this lib, but rather build a new one if we decide to put more effort into the issue around federation upload. The changes made were just done to accommodate the `graphql-upload` upgrade.
